fastcrawler_ui/repository/spiders.py

```diff
@@ -117,23 +117,22 @@ async def update_task(
         self.crawlers.add(crawler)
         new_task_settings = task_settings.model_dump()
         task = await self.get_task_by_name(crawler, task_name)
-        if task:
-            new_task: Task = task
+        if task is not None:
             for key, value in new_task_settings.items():
                 if value is not _UNSET:
-                    setattr(new_task, key, value)
+                    setattr(task, key, value)
 
-            return new_task
+            return task
         return None
 
-    async def get_tasks_in_processes(self, crawler: FastCrawler) -> dict[Process, Task]:
+    def get_tasks_in_processes(self, crawler: FastCrawler) -> dict[Process, Task]:
         """Get all tasks from the crawler.
 
         Returns:
             Dict[Process, Task]
         """
         self.crawlers.add(crawler)
-        processes = {process: process.task for process in crawler.crawlers}
+        processes = {process: task for process, task in zip(crawler.crawlers, crawler.controller.app.task_lib.session.tasks)}
         return processes
```

Comment:
get tasks from internal (rocketry) instead of crawlers

Reply:
there is a bug that fixed in  [3c10614](https://github.com/fast-crawler/ui/pull/6/commits/3c10614d59fb1538e1a1654e9c0adff2580220d5)